soc/st/stm32/common/stm32_bitops.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef STM32_BITOPS_H_
+#define STM32_BITOPS_H_
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/arch/common/sys_io.h>
+
+static ALWAYS_INLINE void stm32_set_mask(volatile uint32_t *addr, uint32_t mask)
+{
+	sys_set_bits((mem_addr_t)addr, mask);
+}
+
+static ALWAYS_INLINE void stm32_clear_mask(volatile uint32_t *addr, uint32_t mask)
+{
+	sys_clear_bits((mem_addr_t)addr, mask);
+}
+
+static ALWAYS_INLINE uint32_t stm32_read_mask(volatile uint32_t *addr, uint32_t mask)
+{
+	return (*addr) & mask;
```

Comment:
Use the `sys_read32` primitive:
```suggestion
	return sys_read32((mem_addr_t)addr) & mask;
```

Reply:
Done